src/bootstrap/util.rs

```diff
@@ -30,6 +30,29 @@ pub fn staticlib(name: &str, target: &str) -> String {
     }
 }
 
+#[cfg(not(windows))]
+pub fn build_path(path: &str) -> PathBuf {
+    PathBuf::from(path)
+}
+
+#[cfg(windows)]
+pub fn build_path(path: &str) -> PathBuf {
```

Comment:
This is somewhat innocuously named, but perhaps it could be contained to just `config.rs` which should be the only source of these paths (reading them from `config.mk` which is generated from a shell script, `configure`).

Perhaps this could be called `msys_path_to_system(&path)` as well?